### ISSUE TITLE
current Cyberduck version - uninstall issue

### ISSUE THREAD

**Issue by scheste** [saltstack/salt-winrepo-ng#1540]
Hi,

I want to update the Cyberduck state (salt version 2018.3.3 + Windows Server 2019 Client), because newer versions also as MSI package available and the uninstall procedur has been changed
But I don't understand why only the installation works and not also the uninstall as well. I have tried it with various MSI uninstall parameters, unfortunately always with no success. With the exe installer I have the same problem.

```
cyberduck:
{% for major, subversions in versions.items() %}
{% for minor in subversions %}
  '{{major}}.{{minor}}':
    full_name: 'Cyberduck {{major}}'
    installer: '{{source_path}}/Cyberduck-Installer-{{major}}.{{minor}}.msi'
    uninstaller: '{{source_path}}/Cyberduck-Installer-{{major}}.{{minor}}.msi'
    install_flags: '/qn'
    uninstall_flags: '/qn'
    msiexec: True
    locale: en_US
    reboot: False
{% endfor %}
{% endfor %}
```

Console Output
```
> salt 'win2019' pkg.install cyberduck
win2019:
    ----------
    Cyberduck:
        ----------
        new:
            6.9.0.29768
        old:
    cyberduck:
        ----------
        install status:
            success
salt 'win2019' pkg.remove cyberduck
win2019:
    ----------
    cyberduck:
        ----------
        current:
            not installed
```

**Comment by twangboy** [comment]
You're missing the uninstaller parameter

**Comment by scheste** [comment]
I have tried also with uninstaller parameter
`uninstaller: '{{source_path}}/Cyberduck-Installer-{{major}}.{{minor}}.msi'`
and 
`uninstaller: 'msiexec.exe'`
(also with path variant)

When I uninstall manually via CMD and the following parameters:
`msiexec.exe /x {4DB85437-9B91-4E68-8019-EE31C4464DA4} /qn`
it works, but not with salt.
Do I have to consider something with a 32bit msi package under Windows x64?

**Comment by twangboy** [comment]
I wonder if it's the GUID that's messing things up. The other pkg defs that handle MSI's use the same path and flags as the installer....
```
uninstaller: '{{source_path}}/Cyberduck-Installer-{{major}}.{{minor}}.msi'
uninstall_flags: /qn /norestart
```

**Comment by twangboy** [comment]
Does the package show up in `pkg.list_pkgs` after it's installed? How does that return look?

**Comment by scheste** [comment]
Cyberduck will be listed after the installation and even after I tried to uninstall it:

`Cyberduck:
        6.9.0.29768`

I have test it also with 2 older build versions, the same problem.
Maybe the Cyberduck installer has a bug too...
I can uninstall the package via the CMD or cmd.run also with parameter:

`MsiExec.exe /x c:\Users\Administrator\Downloads\Cyberduck-Installer-6.9.0.29768.msi /qn`
or
`msiexec.exe /x {4DB85437-9B91-4E68-8019-EE31C4464DA4} /qn`

I have tested also the 7zip package on this test system and this works as expected.

salt 'win2019' pkg.install 7zip
win2019:
    ----------
    7zip:
        ----------
        new:
            18.06.00.0
        old:
salt 'win2019' pkg.remove 7zip
win2019:
    ----------
    7zip:
        ----------
        new:
        old:
            18.06.00.0`

The only difference is that 7zip is also installed in the 64bit variant, cyberduck is only available as 32bit.

**Comment by scheste** [comment]
The cyberduck exe installer has exactly the same problem, installation works, uninstall not.

`{% set versions = {'6.9.0':['29768']} %}
{% set source_path = 'https://update.cyberduck.io/windows' %}

cyberduck:
{% for major, subversions in versions.items() %}
{% for minor in subversions %}
  '{{major}}.{{minor}}':
    full_name: 'Cyberduck {{major}}'
    installer: '{{source_path}}/Cyberduck-Installer-{{major}}.{{minor}}.exe'
    uninstaller: '{{source_path}}/Cyberduck-Installer-{{major}}.{{minor}}.exe'
    install_flags: '/install /quiet'
    uninstall_flags: '/uninstall /quiet'
    msiexec: False
    locale: en_US
    reboot: False
{% endfor %}
{% endfor %}`

Install and uninstall in the CMD works with the same parameters.

**Comment by twangboy** [comment]
The problem might be the `full_name`:
```
    full_name: 'Cyberduck {{major}}'
```
It looks like the full name might be just `Cyberduck` with no version. When you look at `Add/Remove Programs` does it show Cyberduck or Cyberduck and some version in the Name column?
That's what needs to be in the `full_name` to match correctly.
My guess is that you need to remove the `{{major}}`

**Comment by scheste** [comment]
thanks, now it works, sorry for your inconvenience. I have used and modified the current state, in the former version there was still a version specification in the name. So I overlooked, lesson learned...

Important is to use a higer timeout instead of the standard values, because the installation and deinstallation generates a high cpu utilization. Otherwise I get the message `Minion did not return. [No response]`
thanks again!
